sdk/core/azure_core_amqp/src/value.rs

```diff
@@ -267,10 +304,14 @@ where
         self.inner.push((key, value));
     }
 
-    pub fn get(&self, key: &K) -> Option<&V> {
+    pub fn get<Q>(&self, key: &Q) -> Option<&V>
+    where
+        K: Borrow<Q> + PartialEq<Q>,
+        Q: PartialEq<K> + ?Sized,
+    {
         self.inner
             .iter()
-            .find_map(|(k, v)| if k == key { Some(v) } else { None })
+            .find_map(|(k, v)| if key == k { Some(v) } else { None })
```

Comment:
Ah, this may be why you had to implement both directions. As I [changed in my example](https://gist.github.com/heaths/dd2a8f327b1a6b429827ecc17fa44460/revisions#diff-0544fc952eefa8d170bbd200b75827f5bb58e39b132e7c871c1bc8a9d5e09ce5), if you are careful of the order you only need the direction you use, and can use it both ways.

That said, again though, if the goal is to make it symmetric for other callers, by all means keep it. It just puts more constraints on using your ordered map.

Reply:
Actually, even if you keep the symmetrical implementations, I would still advise you to maintain a single order - maybe even call `eq` directly just to avoid ambiguity for future maintainers - so that both implementations aren't compiled into a binary. You know, shave a little off the top.

Reply:
I like the .eq idea, as long as VS Code doesn't decide to change it to == for me :(.